src/gluonts/mx/distribution/mixture.py

```diff
@@ -77,12 +77,18 @@ def __init__(
                                     the axis)."""
 
             expected_shape = self.batch_shape + (len(components),)
-            assert len(expected_shape) == len(mixture_probs.shape), (
+
+            # fix edge case: if batch_shape == (1,) the mixture_probs shape is squeezed to (k,)
+            # reshape it to (1, k)
+            if self.batch_shape == (1,):
```

Comment:
When does this squeeze happen? Do you want to account for invalid user input for that edge case? I think the documentation is quite clear about the shape and in any case I added the assertion to make sure that the mixture_probs have the correct shape.
For a batch shape of (,1) this is the correct way of defining the mixture_probs:
`mixture_probs=mx.nd.array([[0.1, 0.9]]) `
while this is not:
`mixture_probs=mx.nd.array([0.1, 0.9]) `

Also this is currently in the 
`if not isinstance(mixture_probs, mx.sym.Symbol):`
block, which probably is not intended? The block is there because the assertions do not work with mx.sym.Symbol mode.

Reply:
I think this squeeze is happening when we get a subset of the batch_shape distribution and is due to this:
https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/mx/distribution/distribution.py#L43

Assuming a 2D batch_shape, If I cut just a single distribution, i.e., `distr[i, j]`, then the `mixture_probs` will be of shape `(k,)` and not `(1,k)`. I guess when you define this distribution there is no issue but when you cut it there is.

I will move it outside of the block. Good catch, thanks!

Reply:
I actually moved the fix to `__getitem__` which is the method that is not consistent. I guess this addresses both of your comments simultaneously.

Reply:
I just thought about this way of addressing it, too. Makes sense! 👍